src/main/java/dev/gymService/service/interfaces/UserService.java

```
@@ -0,0 +1,21 @@
+package dev.gymService.service.interfaces;
+
+import dev.gymService.model.User;
+import dev.gymService.repository.implmentations.TraineeRepositoryImpl;
+import dev.gymService.utills.FileLogger;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public interface UserService {
+    static final Logger logger = FileLogger.getLogger(TraineeRepositoryImpl.class);
+    default boolean isAuthenticated(String userName, String password, User user) {
+        if (user != null && user.getUserName().equals(userName) && user.getPassword().equals(password)) {
+            logger.log(Level.INFO, "Successful authentication for trainee: " + userName);
+            return true;
+        } else {
+            logger.log(Level.INFO, "Successful authentication for trainee: " + userName);
```

1. the same log as on #14, need to adjust
2. I would suggest throwing exception instead of returning boolean

---

Done. For failed authentication scenario, the logic was corrected to log proper log message and throw runtime exception.